### PR TITLE
ha: refactor online DDL

### PR DESCRIPTION
**Pull request by csuzhangxc**
### What problem does this PR solve? 

refactor online DDL feature to be compatible with HA.

### What is changed and how it works?

record source ID instead of server ID in online DDL meta table.

### Check List 

Tests 

 - Unit test
 - Integration test

Code changes

 - Has persistent data change

Side effects

 - Breaking backward compatibility

Related changes

 - Need to be included in the release note

### CODE REVIEW

**Comment by csuzhangxc** [comment]
@WangXiangUSTC @lichunzhu PTAL

File: tests/online_ddl/run.sh [hunk 1]
```diff
@@ -33,15 +33,35 @@ function real_run() {
...
 
-    # use sync_diff_inspector to check data now!
-    # check_sync_diff $WORK_DIR $cur/conf/diff_config.toml
+    echo "use sync_diff_inspector to check increment data"
+    check_sync_diff $WORK_DIR $cur/conf/diff_config.toml
+
+    echo "start dm-worker3 and kill dm-worker2"
+    ps aux | grep dm-worker2 |awk '{print $2}'|xargs kill || true
+    check_port_offline $WORKER2_PORT 20
+    rm -rf $WORK_DIR/worker2/relay_log
```
Comment (WangXiangUSTC) [inline tests/online_ddl/run.sh]: why need remove relay log?

File: tests/online_ddl/run.sh [hunk 2]
```diff
@@ -33,15 +33,35 @@ function real_run() {
...
+
+    run_dm_worker $WORK_DIR/worker3 $WORKER3_PORT $cur/conf/dm-worker3.toml
+    check_rpc_alive $cur/../bin/check_worker_online 127.0.0.1:$WORKER3_PORT
+
+    sleep 8
+    echo "wait and check task running"
+    run_dm_ctl_with_retry $WORK_DIR "127.0.0.1:$MASTER_PORT" \
```
Comment (WangXiangUSTC) [inline tests/online_ddl/run.sh]: seems already had retry, don't need to wait 8 seconds in L53?

File: tests/online_ddl/run.sh (hunk 1 above)
Comment (csuzhangxc) [reply tests/online_ddl/run.sh]: no necessary, revert in https://github.com/pingcap/dm/pull/494/commits/4a5e461ef8ce96ed4d4c034b15b53a2a8bd48027.

File: tests/online_ddl/run.sh (hunk 2 above)
Comment (csuzhangxc) [reply tests/online_ddl/run.sh]: removed in https://github.com/pingcap/dm/pull/494/commits/4a5e461ef8ce96ed4d4c034b15b53a2a8bd48027.

**Pull request merged**